### PR TITLE
[receiver/kafka] Add JSON-encoded log support

### PR DESCRIPTION
**Pull request by Caleb-Hurshman**
**Description:** <Describe what has changed.>
Added support for json encoded logs to the `kafkareceiver`

**Link to tracking Issue:** #20734

**Testing:** Added limited tests for `json_unmarshaler.go`, similar to existing 'unmarshaler' files.

**Documentation:** Updated `README.md` of the `kafkareceiver` to include the JSON encoding option.

### CODE REVIEW

**Review by djaglowski — COMMENTED** [review]
Looks reasonable to me but codeowners @pavolloffay and @MovieStoreGuy may have more context.

**Comment by Caleb-Hurshman** [comment]
@MovieStoreGuy @pavolloffay Re-pinging for review, need feedback on this `kafakreceiver` enhancement

**Comment by Caleb-Hurshman** [comment]
@MovieStoreGuy @pavolloffay Re-pinging for review, need feedback on this kafakreceiver enhancement

**Review by djaglowski — APPROVED** [review]
LGTM.

@Caleb-Hurshman, the lint failure appears unrelated but I can't get the job to cancel. I think rebasing may help.

I would appreciate a review from @MovieStoreGuy or @pavolloffay but I think this is straightforward enough to merge if they do not have an opinion.

File: receiver/kafkareceiver/json_unmarshaler_test.go [hunk 1]
```diff
@@ -0,0 +1,16 @@
...
+	"testing"
+
+	"github.com/stretchr/testify/assert"
+)
+
+func TestNewJSONUnmarshaler(t *testing.T) {
+	t.Parallel()
+	um := newJSONLogsUnmarshaler()
+	assert.Equal(t, "json", um.Encoding())
+}
```
Comment (MovieStoreGuy) [inline receiver/kafkareceiver/json_unmarshaler_test.go:16]: Do you mind adding some tests to validate that a given json blob of `{blah ...}` is encoding as a plog value?

File: receiver/kafkareceiver/json_unmarshaler.go [hunk 2]
```diff
@@ -0,0 +1,43 @@
...
+	return &jsonLogsUnmarshaler{}
+}
+
+func (r *jsonLogsUnmarshaler) Unmarshal(buf []byte) (plog.Logs, error) {
+	// create a new Logs struct to be populated with log data and returned
+	p := plog.NewLogs()
+
+	// get json logs from the buffer
+	jsonVal := map[string]interface{}{}
+	if err := json.Unmarshal(buf, &jsonVal); err != nil {
```
Comment (MovieStoreGuy) [inline receiver/kafkareceiver/json_unmarshaler.go:26]: Could I ask you to use `jsoniter` instead?

It has improved performance over the standard lib json encoding.

**Review by MovieStoreGuy — COMMENTED** [review]
Sorry for missing pings, I have a couple comments that just to address.

The only thing blocking on my behalf is adding tests just validate marshalling.

Are you also able to add this to the kafka exporters.

**Comment by Caleb-Hurshman** [comment]
@MovieStoreGuy this enhancement was specifically assigned for the `kafkareceiver`. I may revisit this in the future, but for now I will _personally_ not be implementing this for the `kafkaexporter`.

**Comment by djaglowski** [comment]
@Caleb-Hurshman, please address CI failures

**Review by djaglowski — APPROVED** [review]

**Comment by Caleb-Hurshman** [comment]
@MovieStoreGuy CI is passing now, ready for your review

**Review by codeboten — COMMENTED** [review]
something looks like it's gone wonky w/ the rebase, it's showing 242 commits

File: receiver/kafkareceiver/go.mod [hunk 3]
```diff
@@ -44,7 +45,10 @@ require (
 	github.com/jcmturner/gokrb5/v8 v8.4.3 // indirect
 	github.com/jcmturner/rpc/v2 v2.0.3 // indirect
 	github.com/jmespath/go-jmespath v0.4.0 // indirect
+<<<<<<< HEAD
 	github.com/json-iterator/go v1.1.12 // indirect
+=======
+>>>>>>> 4e7801ca7266cd02870e770e1b8917b35059f244
```
Comment (atoulme) [inline receiver/kafkareceiver/go.mod:51]:
```suggestion
	github.com/json-iterator/go v1.1.12 // indirect
```

**Pull request closed without merging**